bin/disco_aws.py

```diff
@@ -266,29 +296,35 @@ def run():
             ami_dict = bake.list_amis_by_instance(instances)
             now = datetime.utcnow()
 
+        header = format_listhosts_header(args, most)
+        divider = '-'.ljust(len(header), '-')
+
+        print("\n" + header + "\n" + divider)
+
         for instance in instances_sorted:
-            line = u"{0} {1:<30} {2:<15}".format(
+            line = u"{0:<11} {1:<25} {2:<15}".format(
                 instance.id, instance.tags.get("hostclass", "-"),
                 instance.ip_address or instance_to_private_ip[instance.id])
             if args.state or most:
                 line += u" {0:<10}".format(instance.state)
             if args.hostname or most:
-                line += u" {0:<1}".format("-" if instance.tags.get("hostname") is None else "y")
+                line += u" {0:<30}".format(
+                    "-" if instance.tags.get("hostname") is None else instance.tags.get("hostname"))
```

Comment:
or `...format(instance.tags.get("hostname", "-"))` for a slightly different impact—are we concerned about an empty-string hostname tag?